server/src/test/java/io/druid/curator/discovery/ServerDiscoverySelectorTest.java

```diff
@@ -52,14 +53,62 @@ public void testPick() throws Exception
     EasyMock.expect(serviceProvider.getInstance()).andReturn(instance).anyTimes();
     EasyMock.expect(instance.getAddress()).andReturn(ADDRESS).anyTimes();
     EasyMock.expect(instance.getPort()).andReturn(PORT).anyTimes();
-    EasyMock.replay(instance,serviceProvider);
+    EasyMock.replay(instance, serviceProvider);
     Server server = serverDiscoverySelector.pick();
-    Assert.assertEquals(PORT,server.getPort());
-    Assert.assertEquals(ADDRESS,server.getAddress());
+    Assert.assertEquals(PORT, server.getPort());
+    Assert.assertEquals(ADDRESS, server.getAddress());
     Assert.assertTrue(server.getHost().contains(new Integer(PORT).toString()));
     Assert.assertTrue(server.getHost().contains(ADDRESS));
     Assert.assertEquals(new String("http"), server.getScheme());
-    EasyMock.verify(instance,serviceProvider);
+    EasyMock.verify(instance, serviceProvider);
+    final URI uri = new URI(String.format("%s://%s%s", server.getScheme(), server.getHost(), "/druid/indexer/v1/action"));
```

Comment:
since this PR is about removing potentially bad String.format assumptions, let's use `new URI(scheme, host, path, null)` instead of String.format here :)